File: devcamp2018/app/src/main/java/com/psato/devfest/di/module/NetworkModule.java
package com.psato.devfest.di.module;

import com.psato.devfest.data.remote.TraktAPI;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;

/**
 * Created by athila on 31/05/16.
 */

@Module
public class NetworkModule {

    @Provides
    @Singleton
    TraktAPI provideUserApi(Retrofit retrofit) {
        return retrofit.create(TraktAPI.class);
    }
}

File: devcamp2018/app/src/main/java/com/psato/devcamp/di/module/NetworkModule.kt
package com.psato.devcamp.di.module

import com.psato.devcamp.data.remote.TraktAPI

import javax.inject.Singleton

import dagger.Module
import dagger.Provides
import retrofit2.Retrofit

/**
 * Created by athila on 31/05/16.
 */

@Module
class NetworkModule {

    @Provides
    @Singleton
    internal fun provideUserApi(retrofit: Retrofit): TraktAPI {
        return retrofit.create(TraktAPI::class.java)
    }
}
